enclave-manager/web/packages/app/src/emui/enclaves/components/widgets/ImageButton.tsx

```diff
@@ -109,17 +102,36 @@ export const SetImageModel = ({ isOpen, onClose, currentImage, serviceName, encl
     }
 
     const packageId = starlarkRun.value.packageId;
+    const packageInfoResponse = await getPackageInfo(packageId);
+    if (packageInfoResponse.isErr) {
+      setError(`Error occurred getting info about ${packageId} from indexer.`);
+      return;
+    }
+    if (!packageInfoResponse.value.package) {
+      setError(`Could not find package ${packageId}`);
+      return;
+    }
+    const packageArgs = packageInfoResponse.value.package.args;
+
+    console.log(`initial serialized params: ${starlarkRun.value.initialSerializedParams}`);
+    if (!starlarkRun.value.initialSerializedParams) {
+      setError(`Error occurred getting initial params used to start package.`);
+      return;
+    }
+    const argsRecord = JSON.parse(starlarkRun.value.initialSerializedParams);
+    if (typeof argsRecord !== "object" || argsRecord === null) {
+      setError("Error: deserializing initial params of starlark package run failed.");
+      return;
+    }
 
-    let argsRecord;
-    if (starlarkRun.value.initialSerializedParams) {
-      console.log(`initial serialized params: ${starlarkRun.value.initialSerializedParams}`);
-      argsRecord = deserializeParams(starlarkRun.value.initialSerializedParams);
+    let args;
+    console.log(packageArgs);
+    if (packageArgs.length === 2 && packageArgs[1].name === "args") {
+      args = objectToStarlark(wrapWithArgs(argsRecord), 4);
```

Comment:
if the package is not one that uses `kwargs` and instead still uses the old method of passing in an `args` object, we have to wrap the `serializedParams` in an `args` object. i do this by retrieving info about a packages `args` from the packageIndexer and checking it only has two params and one of them is the `args` object.